trace/obfuscate.go

```diff
@@ -1,19 +1,61 @@
 package trace
 
 import (
+	"regexp"
+
 	pb "github.com/akitasoftware/akita-ir/go/api_spec"
 	"github.com/akitasoftware/akita-libs/spec_util"
 	. "github.com/akitasoftware/akita-libs/visitors"
 	vis "github.com/akitasoftware/akita-libs/visitors/http_rest"
 	"github.com/postmanlabs/postman-insights-agent/printer"
 )
 
-func obfuscate(m *pb.Method) {
-	var ov obfuscationVisitor
-	vis.Apply(&ov, m)
+type Obfuscator struct {
+	SensitiveDataKeys          []string
+	SensitiveDataValuePatterns []*regexp.Regexp
+}
+
+func NewObfuscator() *Obfuscator {
+	// Load obfuscation config using relative path to load the config file.
+	config, err := configFromFile("obfuscation_config.yaml")
+	if err != nil {
+		printer.Errorf("failed to load obfuscation config: %v\n", err)
+		return nil
+	}
 
-	// Mark the method as obfuscated.
-	m.GetMeta().GetHttp().Obfuscation = pb.HTTPMethodMeta_ZERO_VALUE
+	sensitiveDataRegex := make([]*regexp.Regexp, len(config.SensitiveValueRegexes))
+
+	// Compile regex patterns.
+	for i, pattern := range config.SensitiveValueRegexes {
+		regex, err := regexp.Compile(pattern)
+		if err != nil {
+			printer.Errorf("failed to compile regex pattern %s: %v\n", pattern, err)
+			return nil
+		}
+		sensitiveDataRegex[i] = regex
+	}
+
+	return &Obfuscator{
+		SensitiveDataKeys:          config.SensitiveKeys,
+		SensitiveDataValuePatterns: sensitiveDataRegex,
+	}
+}
+
+func (o *Obfuscator) obfuscate(m *pb.Method, obfuscateWholePayload bool) {
```

Comment:
I don't think there's any reason to make this method unexported?
```suggestion
func (o *Obfuscator) Obfuscate(m *pb.Method, obfuscateWholePayload bool) {
```